checks/evaluation/vulnerabilities.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluation
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+)
+
+// Vulnerabilities applies the score policy for the Vulnerabilities check.
+func Vulnerabilities(name string, dl checker.DetailLogger,
+	r *checker.VulnerabilitiesData) checker.CheckResult {
+	if r == nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	score := checker.MaxResultScore
+	IDs := []string{}
+	for _, vuln := range r.Vulnerabilities {
+		IDs = append(IDs, vuln.ID)
+		score--
+	}
+
+	if score < 0 {
+		score = 0
+	}
+
+	if len(IDs) > 0 {
+		dl.Warn3(&checker.LogMessage{
+			Text: fmt.Sprintf("HEAD is vulnerable to %s", strings.Join(IDs, ", ")),
```

Comment:
Could we extract the SHA hear and say something like:

`HEAD (%s) is vuln..., SHA`

Reply:
Good idea, will do.

Reply:
mhh, actually the data is already available as part of the overall scorecard results:
```
{
  "date": "2022-01-26",
  "repo": {
    "name": "github.com/org/name",
    "commit": "de6367caa31b59e2156f83b04c2f30611b7ac393"
  },
vulnerabilities{...}
```

So I'll keep as-is. Please re-open if you disagree.

Reply:
Fine to keep as-is, I think!
Just as long as the data is there to grab if we need it later :)